### DISCUSSION TITLE
Has anyone successfully deployed on Google Cloud?

### DISCUSSION THREAD

**Discussion by hiyukoim** [danny-avila/LibreChat#3922] [Troubleshooting]
I've been working hard on getting Docker up and running on a Google Cloud e2-micro instance. I followed all the steps on this page (https://www.librechat.ai/docs/user_guides) and everything seemed to be going smoothly - all the Docker containers are running just fine. However, when I try to access the external IP address, I'm getting a "502 bad gateway" error for HTTP and no response for HTTPS.

I've checked the error logs and added firewall rules, but no luck so far. I have a feeling there might be a Google Cloud-specific task I'm missing.

![IMG_5194](https://github.com/user-attachments/assets/bf26ee93-d7a8-4008-912d-e899b9bf8883)
![IMG_5195](https://github.com/user-attachments/assets/9279cdd0-45b4-441c-bce8-22c00024447a)

If anyone has experience with this, I'd really appreciate a helping hand. I'm really eager to get this up and running smoothly. Thanks in advance!

**Comment by Andrew-Chen-Wang** [comment]
I've successfully deployed without issue, though I didn't use the deploy YAML configuration and preferred deploying things myself via Pulumi (I just vibe coded things) and used Cloud Run. I had to create my own Dockerhub image, though, since ghrc is not allowed in Cloud Run, only dockerhub or GCP registry.

**Comment by Ziyann** [comment]
I also deployed on Cloud Run, building my own image to Artifact Registry. Nothing complicated, like 5 lines inheriting from the LibreChat image and adding librechat.yaml and such. For MongoDB, using Firestore Enterprise with MongoDB compatibility didn't work, so I used MongoDB Atlas, also available from Google Cloud Marketplace with a generous free tier.
Request based instances with 1 min/max instance (to avoid cold starts) seem to work fine so far, so the whole setup is almost free.

One thing I miss is that LibreChat doesn't support Application Default Credentials (ADC), to automatically fetch an access token from the metadata server, without having to specify any keys or service account credentials to use Vertex AI. LangChain supports this.
